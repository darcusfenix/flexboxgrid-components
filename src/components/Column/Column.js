// @flow
import * as React from "react"

import classNames from "classnames"
import {
    prefixColumn,
    xsConst,
    smConst,
    mdConst,
    lgConst,
    prefixOffset,
    reorderingFirst,
    reorderingLast
} from "../constants"

type Props = {
    xsAuto?: boolean,
    smAuto?: boolean,
    mdAuto?: boolean,
    lgAuto?: boolean,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    xsOffset?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    smOffset?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    mdOffset?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    lgOffset?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    xs?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    sm?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    md?: number | string,
    /** Any value of 1,2,3,4,5,6,7,8,9,10,11,12*/
    lg?: number | string,
    xsFirst?: boolean,
    smFirst?: boolean,
    mdFirst?: boolean,
    lgFirst?: boolean,
    xsLast?: boolean,
    smLast?: boolean,
    mdLast?: boolean,
    lgLast?: boolean,
    children?: any
}

function Column({
    xsOffset,
    smOffset,
    mdOffset,
    lgOffset,
    xs,
    sm,
    md,
    lg,
    xsAuto,
    smAuto,
    mdAuto,
    lgAuto,
    xsFirst,
    smFirst,
    mdFirst,
    lgFirst,
    xsLast,
    smLast,
    mdLast,
    lgLast,
    children
}: Props) {
    const columnClass = classNames(
        xsAuto && `${prefixColumn}-${xsConst}`,
        smAuto && `${prefixColumn}-${smConst}`,
        mdAuto && `${prefixColumn}-${mdConst}`,
        lgAuto && `${prefixColumn}-${lgConst}`,
        xsOffset && `${prefixColumn}-${xsConst}-${prefixOffset}-${xsOffset}`,
        smOffset && `${prefixColumn}-${smConst}-${prefixOffset}-${smOffset}`,
        mdOffset && `${prefixColumn}-${mdConst}-${prefixOffset}-${mdOffset}`,
        lgOffset && `${prefixColumn}-${lgConst}-${prefixOffset}-${lgOffset}`,
        xs && `${prefixColumn}-${xsConst}-${xs}`,
        sm && `${prefixColumn}-${smConst}-${sm}`,
        md && `${prefixColumn}-${mdConst}-${md}`,
        lg && `${prefixColumn}-${lgConst}-${lg}`,
        xsFirst && `${reorderingFirst}-${xsConst}`,
        smFirst && `${reorderingFirst}-${smConst}`,
        mdFirst && `${reorderingFirst}-${mdConst}`,
        lgFirst && `${reorderingFirst}-${lgConst}`,
        xsLast && `${reorderingLast}-${xsConst}`,
        smLast && `${reorderingLast}-${smConst}`,
        mdLast && `${reorderingLast}-${mdConst}`,
        lgLast && `${reorderingLast}-${lgConst}`
    )

    return <div className={columnClass}>{children}</div>
}

Column.defaultProps = {
    xsAuto: false,
    smAuto: false,
    mdAuto: false,
    lgAuto: false,
    xsFirst: false,
    smFirst: false,
    mdFirst: false,
    lgFirst: false,
    xsLast: false,
    smLast: false,
    mdLast: false,
    lgLast: false
}

export default Column
