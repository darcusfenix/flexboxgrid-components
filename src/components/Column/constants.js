export const PROPS_VIEWPORT_WIDTH = {
    xs: 12,
    sm: 12,
    md: 12,
    lg: 12
}

export const PROPS_VIEWPORT_WIDTH_AUTO = {
    xsAuto: true,
    smAuto: true,
    mdAuto: true,
    lgAuto: true
}

export const PROPS_VIEWPORT_WIDTH_OFFSET = {
    xsOffset: true,
    smOffset: true,
    mdOffset: true,
    lgOffset: true
}

export const PROPS_VIEWPORT_WIDTH_FIRST = {
    xsFirst: true,
    smFirst: true,
    mdFirst: true,
    lgFirst: true
}

export const PROPS_VIEWPORT_WIDTH_LAST = {
    xsLast: true,
    smLast: true,
    mdLast: true,
    lgLast: true
}
