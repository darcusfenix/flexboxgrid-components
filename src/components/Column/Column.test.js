import React from "react"
import { shallow } from "enzyme"

import Column from "./Column"
import {
    PROPS_VIEWPORT_WIDTH,
    PROPS_VIEWPORT_WIDTH_AUTO,
    PROPS_VIEWPORT_WIDTH_FIRST,
    PROPS_VIEWPORT_WIDTH_LAST,
    PROPS_VIEWPORT_WIDTH_OFFSET
} from "./constants"

describe("<Column />", () => {
    it("should render correctly", () => {
        const tree = shallow(<Column />)
        expect(tree).toMatchSnapshot()
    })

    it("should render correctly with all props of viewport width", () => {
        const tree = shallow(<Column {...PROPS_VIEWPORT_WIDTH} />)
        expect(tree).toMatchSnapshot()
    })

    it("should render correctly with all props of viewport width Auto", () => {
        const tree = shallow(<Column {...PROPS_VIEWPORT_WIDTH_AUTO} />)
        expect(tree).toMatchSnapshot()
    })

    it("should render correctly with all props of viewport width Offset", () => {
        const tree = shallow(<Column {...PROPS_VIEWPORT_WIDTH_OFFSET} />)
        expect(tree).toMatchSnapshot()
    })

    it("should render correctly with all props of viewport width First", () => {
        const tree = shallow(<Column {...PROPS_VIEWPORT_WIDTH_FIRST} />)
        expect(tree).toMatchSnapshot()
    })

    it("should render correctly with all props of viewport width Last", () => {
        const tree = shallow(<Column {...PROPS_VIEWPORT_WIDTH_LAST} />)
        expect(tree).toMatchSnapshot()
    })
})
