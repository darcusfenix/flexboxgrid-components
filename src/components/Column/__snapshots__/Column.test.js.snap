// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lgAuto={false}
    lgFirst={false}
    lgLast={false}
    mdAuto={false}
    mdFirst={false}
    mdLast={false}
    smAuto={false}
    smFirst={false}
    smLast={false}
    xsAuto={false}
    xsFirst={false}
    xsLast={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Column /> should render correctly with all props of viewport width 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lg={12}
    lgAuto={false}
    lgFirst={false}
    lgLast={false}
    md={12}
    mdAuto={false}
    mdFirst={false}
    mdLast={false}
    sm={12}
    smAuto={false}
    smFirst={false}
    smLast={false}
    xs={12}
    xsAuto={false}
    xsFirst={false}
    xsLast={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "col-xs-12 col-sm-12 col-md-12 col-lg-12",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Column /> should render correctly with all props of viewport width Auto 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lgAuto={true}
    lgFirst={false}
    lgLast={false}
    mdAuto={true}
    mdFirst={false}
    mdLast={false}
    smAuto={true}
    smFirst={false}
    smLast={false}
    xsAuto={true}
    xsFirst={false}
    xsLast={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "col-xs col-sm col-md col-lg",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "col-xs col-sm col-md col-lg",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Column /> should render correctly with all props of viewport width First 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lgAuto={false}
    lgFirst={true}
    lgLast={false}
    mdAuto={false}
    mdFirst={true}
    mdLast={false}
    smAuto={false}
    smFirst={true}
    smLast={false}
    xsAuto={false}
    xsFirst={true}
    xsLast={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "first-xs first-sm first-md first-lg",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "first-xs first-sm first-md first-lg",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Column /> should render correctly with all props of viewport width Last 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lgAuto={false}
    lgFirst={false}
    lgLast={true}
    mdAuto={false}
    mdFirst={false}
    mdLast={true}
    smAuto={false}
    smFirst={false}
    smLast={true}
    xsAuto={false}
    xsFirst={false}
    xsLast={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "last-xs last-sm last-md last-lg",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "last-xs last-sm last-md last-lg",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Column /> should render correctly with all props of viewport width Offset 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    lgAuto={false}
    lgFirst={false}
    lgLast={false}
    lgOffset={true}
    mdAuto={false}
    mdFirst={false}
    mdLast={false}
    mdOffset={true}
    smAuto={false}
    smFirst={false}
    smLast={false}
    smOffset={true}
    xsAuto={false}
    xsFirst={false}
    xsLast={false}
    xsOffset={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "col-xs-offset-true col-sm-offset-true col-md-offset-true col-lg-offset-true",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "col-xs-offset-true col-sm-offset-true col-md-offset-true col-lg-offset-true",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
